core/api_threads.go

```diff
@@ -121,3 +122,86 @@ func (a *api) rmThreads(g *gin.Context) {
 	}
 	g.String(http.StatusOK, "ok")
 }
+
+func (a *api) streamThreads(g *gin.Context) {
+	id := g.Param("id")
+	thrd := a.node.Thread(id)
+	if thrd == nil {
+		g.String(http.StatusNotFound, "thread not found")
+		return
+	}
+	opts, err := a.readOpts(g)
+	if err != nil {
+		a.abort500(g, err)
+		return
+	}
+
+	listener := a.node.ThreadUpdateCh()
+	g.Stream(func(w io.Writer) bool {
+		select {
+		case update, ok := <-listener.Ch:
+			if !ok {
+				return false
+			}
+			if data, ok := update.(ThreadUpdate); ok {
+				info, _ := addBlockInfo(a, data)
+				if opts["events"] == "true" {
+					g.SSEvent("threadUpdate", info)
+				} else {
+					g.JSON(http.StatusOK, info)
+				}
+			}
+		default:
+		}
+		return true
+	})
+
+	listener.Close()
+}
+
+func addBlockInfo(a *api, update ThreadUpdate) (ThreadUpdate, error) {
+	block := update.Block
+	username := a.node.ContactUsername(block.AuthorId)
+
+	var info interface{}
+	switch update.Block.Type {
+	case repo.FilesBlock:
+		info, _ = a.node.threadFile(&update.Block)
+	case repo.CommentBlock:
```

Comment:
seems overly verbose? better way to do this?

Reply:
hhmmm, yeah. see comment below about `BlockInfo`